apis/apps/v1alpha1/opsrequest_webhook.go

```diff
@@ -134,7 +148,18 @@ func (r *OpsRequest) validateClusterPhase(cluster *Cluster) error {
 	// check if the opsRequest can be executed in the current cluster phase unless this opsRequest is reentrant.
 	if !slices.Contains(opsBehaviour.FromClusterPhases, cluster.Status.Phase) &&
 		!slices.Contains(opsNamesInQueue, r.Name) {
-		return fmt.Errorf("OpsRequest.spec.type=%s is forbidden when Cluster.status.phase=%s", r.Spec.Type, cluster.Status.Phase)
+		// check if entry-condition is met
+		// if the cluster is not in the expected phase, we should wait for it for up to TTLSecondsBeforeAbort seconds.
+		// if len(opsRecorder) == 0 && !slices.Contains(opsBehaviour.FromClusterPhases, cluster.Status.Phase) {
+		// TTLSecondsBeforeAbort is 0 means that the we do not need to wait for the cluster to reach the expected phase.
+		if r.Spec.TTLSecondsBeforeAbort == nil || (time.Now().After(r.GetCreationTimestamp().Add(time.Duration(*r.Spec.TTLSecondsBeforeAbort) * time.Second))) {
+			return fmt.Errorf("OpsRequest.spec.type=%s is forbidden when Cluster.status.phase=%s", r.Spec.Type, cluster.Status.Phase)
+		}
+		return &WaitForClusterPhaseErr{
+			clusterName:   cluster.Name,
+			currentPhase:  cluster.Status.Phase,
+			expectedPhase: opsBehaviour.FromClusterPhases,
+		}
```

Comment:
if webhook is enabled and returns an error, will blocking the creation of opsRequest.  
should return nil  and check it in the ops_manager.go?

Reply:
Got it. Moved the validation to ops_util.go, and only checks ops in pending phase.